index.html

```diff
@@ -118,16 +205,62 @@
     // TRANSFORM
     txData = data.slice();
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+    if (options.SOCGroup !== 'all') {
+      var SOCGroup=options.SOCGroup;
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+        return d.SuperGroup === SOCGroup;
       });
     }
 
+
     // SCATTERPLOT
 
+    var points=svg.selectAll('.point')
+      .data(txData)
+
+      points.enter().append('circle')
+      .attr('class', 'point')
+      .attr('r', 5)
+
+      points
+      .attr('cx', function(d) { return x(d.WF_Female); })
+      .attr('cy',  function(d) { return y(d.MedEarn); })
+
+
+      points.exit().remove();
+
+
     // HISTOGRAMS
+    var map = data.map(function(d){return d.WF_Female;})
+
+    var histogram = d3.layout.histogram()
+                      .bins(x.ticks())(map);
+
+    var bars = svg.selectAll(".bar")
+                  .data(histogram)
+                  .enter();
+
+   console.log(bars)
+    console.log(bars[0])
+
+     //console.log(d3.extent(bars, function(d) {return d[bars.length]}))
+
+    var yHist = d3.scale.linear()
+      .domain([0,d3.max(bars, function(d) {return d.length})])
+      .range([height, 0])
+      .nice();
+
+    console.log(yHist(51))
+
+
+    bars.append("rect")
+      .attr("width", function(d) { return x(d.dx)/2; })
+      .attr("x", function(d) { return x(d.x)+10; })
```

Comment:
The width is exactly right. But our bars aren't centered, and we're using a magic number (10) here to try to center them. Could we do this just with math? We know:

* The position of the first tick (`x(d.x)')
* The distance to the second rick (`x(d.dx)`)
* The width of each bar (`x(d.dx) / 2`)